src/edu/stanford/nlp/sempre/YearRangeFn.java

```diff
@@ -0,0 +1,60 @@
+package edu.stanford.nlp.sempre;
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Parse year ranges of the form "1990's" or "1800s"
+ *
+ * @author ppasupat
+ */
+public class YearRangeFn extends SemanticFn {
```

Comment:
Would it be better to support DateRanges more generally?  So you can also handle 19th century or 2nd millennium or late 80s?